packages/wallet-lib/src/types/Account/_preparePlugins.js

```diff
@@ -1,9 +1,20 @@
 const sortPlugins = require('./_sortPlugins');
+const logger = require('../../logger');
 
 const preparePlugins = function preparePlugins(account, userUnsafePlugins) {
   function reducer(accumulatorPromise, [plugin, allowSensitiveOperation, awaitOnInjection]) {
     return accumulatorPromise
-      .then(() => account.injectPlugin(plugin, allowSensitiveOperation, awaitOnInjection));
+      .then(async () => {
+        try {
+          await account.injectPlugin(
+            plugin,
+            allowSensitiveOperation,
+            awaitOnInjection,
+          );
+        } catch (e) {
+          logger.error('Error injecting plugin', e);
```

Comment:
Shouldn't we propagate this error to wallet error event?

Reply:
Done